lib/chef/resource/windows_firewall_rule.rb

```diff
@@ -97,8 +110,8 @@ class WindowsFirewallRule < Chef::Resource
         coerce: proc { |d| d.is_a?(String) ? d.split(/\s*,\s*/).sort : Array(d).sort.map(&:to_s) },
         description: "The local port the firewall rule applies to."
 
-      property :remote_address, String,
-        description: "The remote address the firewall rule applies to."
+      property :remote_address, [String, Array],
```

Comment:
Because you coerce the state into an array you also want to coerce the inputs here. Otherwise if a person passes a single string value it won't match the current state value which would be the string in array form. Use coerce here to array and also sort it. Then you don't need to coerce and sort it later when you use it in the action.

Reply:
TIL, thanks!